File: socketd/src/main/java/org/noear/socketd/protocol/Message.java
package org.noear.socketd.protocol;


/**
 * 负载
 *
 * @author noear
 * @since 2.0
 */
public interface Message {

    /**
     * 是否为请求
     */
    boolean isRequest();

    /**
     * 是否为订阅
     */
    boolean isSubscribe();

    /**
     * Key
     */
    String getKey();

    /**
     * topic
     */
    String getTopic();

    /**
     * entity
     */
    Entity getEntity();
}

File: python/socket_websocket/socketd/core/module/Message.py
from abc import ABC, abstractmethod

from .Entity import Entity


class Message(ABC):
    @abstractmethod
    def is_request(self) -> bool:
        pass

    @abstractmethod
    def is_subscribe(self) -> bool:
        pass

    @abstractmethod
    def is_close(self) -> bool:
        pass

    @abstractmethod
    def get_sid(self) -> str:
        pass

    @abstractmethod
    def get_topic(self) -> str:
        pass

    @abstractmethod
    def get_entity(self) -> Entity:
        pass
